src/common/CrosschainBatcher.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {IGateway} from "./interfaces/IGateway.sol";
+import {ICrosschainBatcher} from "./interfaces/ICrosschainBatcher.sol";
+
+import {Auth} from "../misc/Auth.sol";
+
+/// @title  CrosschainBatcher
+/// @dev    Helper contract that enables integrations to automatically batch multiple messages. Should be used like:
+///         ```
+///         contract Integration {
+///             ICrosschainBatcher batcher;
+///
+///             function doSomething(PoolId poolId) external {
+///                 uint256 cost = batcher.batch(abi.encodeWithSelector(Integration._execute.selector, poolId));
```

Comment:
Nit: personally think `batcher.batch` is a bit odd, would prefer `batcher.execute` or `batcher.doCallback` or something else